pkg/storage/cache.go

```diff
@@ -136,3 +143,51 @@ func GetReadStartingWithUserCacheKeyPrefix(store, objectType, relation string) s
 func GetReadCacheKey(store, tuple string) string {
 	return fmt.Sprintf("%sr/%s/%s", iteratorCachePrefix, store, tuple)
 }
+
+// CheckCacheKeyParams is all the necessary pieces to create a unique-per-check cache key.
+type CheckCacheKeyParams struct {
+	StoreID              string
+	AuthorizationModelID string
+	TupleKey             *openfgav1.TupleKey
+	ContextualTuples     []*openfgav1.TupleKey
+	Context              *structpb.Struct
+}
+
+// GetCheckCacheKey converts the elements of a Check into a canonical cache key that can be
+// used for Check resolution cache key lookups in a stable way.
+//
+// For one store and model ID, the same tuple provided with the same contextual tuples and context
+// should produce the same cache key. Contextual tuple order and context parameter order is ignored,
+// only the contents are compared.
+func GetCheckCacheKey(params *CheckCacheKeyParams) (string, error) {
```

Comment:
You improved the benchmark!  🥳 

<img width="1320" alt="image" src="https://github.com/user-attachments/assets/9405d6ff-12e1-4405-a398-afb765a96633">